doc/faq/best_practices.rst

```diff
@@ -0,0 +1,96 @@
+
+.. _best-practices:
+
+**************
+Best Practices
+**************
+
+.. contents::
+   :backlinks: none
+
+
+.. _best-practice-many-figures:
+
+How to save many figures
+========================
+
+You have figured out how to create a plot from your data and either
+show or save the figure, possibly following the tutorial :doc:`/tutorials/introductory/lifecycle`. It looks great and wonderful.::
+
+  data = data_loading_function(fname)
+  fig, ax = plt.subplots()
+  ax.plot(data.x, data.y, label=data.name)
+  ax.legend()
+  ax.set(xlabel='Time [months]', ylabel='Profits [$]', title='Sales Forecast')
+  fig.savefig('sales_forecast.png')
+
+Now you want to make this plot for each of your data files. One might just
+put the above code into a for-loop over all of your filenames. However, you
+will eventually encounter a warning ``More than X figures hae been opened.``.
+With enough looping, Python can eventually run out of memory, and calling
+garbage collection will not help. This is because figures created through
+``pyplot`` will also be stored in within the module.
+
+So, the best practice for saving many figures is to recycle the figure.::
+
+  fig = plt.figure()
+  for fname in data_file_names:
+    data = data_loading_function(fname)
+    ax = fig.subplots()
+    ax.plot(data.x, data.y, label=data.name)
+    ax.legend()
+    ax.set(xlabel='Time [months]', ylabel='Profits [$]', title='Sales Forecast')
+    fig.savefig('sales_forecast_%s.png' % data.name)
+    fig.clear()
+
+Now, only one figure object is ever made, and it is cleaned of any plots
+at each iteration, so the memory usage will not grow.
```

Comment:
... or just call plt.close("all") at the end of the loop.  This avoids having to factor out figure creation from the main loop.

Reply:
What prompted me to make this document was when someone came to me with a script that was growing in memory significantly and they did something like this. I don't remember if they were calling `plt.close()` or not, but recycling the figure definitely solves that problem.